runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -488,7 +488,7 @@ public void processWatermark1(Watermark mark) throws Exception {
       // hold back by the pushed back values waiting for side inputs
       long pushedBackInputWatermark = Math.min(getPushbackWatermarkHold(), mark.getTimestamp());
 
-      timerService.advanceWatermark(pushedBackInputWatermark);
+      timerService.advanceWatermark(pushedBackInputWatermark - 1);
```

Comment:
Yes, this definitely must be documented. However, in other places we don't subtract `1` because the watermark is actually correct: https://github.com/apache/beam/blob/d05246c8681ff6a58c3ed881ff9bcb01963efed9/runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java#L832-L832. We only subtract `1` here to delay firing timers.

Having a method and calling it in some places but not in others can seem quite confusing. What do you think?